Azure/ARO-RP
# pkg/cluster: Use Action for fixupClusterSPObjectID

### Which issue this PR addresses:

[[ARO-1919] AAD: Migrate ARO-RP Azure AD Graph -> Microsoft Graph](https://issues.redhat.com/browse/ARO-1919)

### What this PR does / why we need it:

Minor follow-up to https://github.com/Azure/ARO-RP/pull/1970 where @bennerv requested this in his review.

The `fixupClusterSPObjectID` step in PUCM uses only the [MS Graph client](https://pkg.go.dev/github.com/microsoftgraph/msgraph-sdk-go#GraphServiceClient) (no ARM calls), and the client already handles HTTP 401 Unauthorized errors by refreshing its authorization token. So there's no need to use the [AuthorizationRetryingAction](https://github.com/Azure/ARO-RP/blob/1ee42feb78ae64ca694bf832be3ee17bde8cbea7/pkg/util/steps/refreshing.go#L21-L24) wrapper.
    
The 401 Unauthorized error handling logic for the MS Graph client is in the [github.com/microsoft/kiota-http-go](https://github.com/microsoft/kiota-http-go/blob/main/nethttp_request_adapter.go#L173-L198) module.
   

### Test plan for issue:

- Create an ARO cluster.
- Run `az aro update --refresh-credentials` on it.
- Verify it completes successfully.

### Is there any documentation that needs to be updated for this PR?

No, this is just an internal change.
